### DISCUSSION TITLE
When thread-loader is working, do the those threads read the options of the other loaders?

### DISCUSSION THREAD

**Discussion by nyngwang** [webpack/webpack#18196] [Q&A]
There are some caveats listed in [README.md](https://github.com/webpack-contrib/thread-loader/?tab=readme-ov-file#getting-started) of thread-loader. Specifically, my concern around it is the first rule and the third rule:

> Loaders running in a worker pool are limited. Examples:
> 
> Loaders cannot emit files.
> Loaders cannot use custom loader API (i. e. by plugins).
> Loaders cannot access the webpack options.

### Confusion on Rule 1

Isn't that `babel-loader` also emitting files for webpack eventually? In the example given in the readme, it mentioned that it is expected to work with `babel-loader`. I need some clarifications here as I have some misunderstandings.

### Confusion on Rule 3

In my case, I'm using babel-loader with some custom options like `rootMode: 'upward'`, and in my babel.config.js I also included some third-party plugins like `react-refresh/babel` for HMR with the corresponding plugin in my webpack.config.js. Does this rule mean that `babel-loader` could not access these plugins?

**Comment by alexander-akait** [comment]
> Isn't that babel-loader also emitting files for webpack eventually?

No, files - i.e. assets, emit assets, babel-loader returns code - i.e. string

> Does this rule mean that babel-loader could not access these plugins?

No, it is not related to babel plugins at all

**Reply by nyngwang** [reply]
The first answer is helpful. I'm still not sure whether some of the `options` passing to `babel-loader` would be ignored by thread-loader. I have spent much time reading GitHub Issues on thread-loader but I'm still confused. It seems that for a worker(thread) to access webpack's options, these options need to be serializable. Is this correct? Thank you.

**Reply by alexander-akait** [reply]
All primitive types will be work and serialized

**Reply by nyngwang** [reply]
@alexander-akait I think this should be documented on README.md of thread-loader, I got the same conclusion similar to what you said here by reading those closed/open issues. From the current wording, it says:

> Loaders **cannot** access the webpack options.

Sincerely.

**Reply by alexander-akait** [reply]
Yeah, feel free to sens a PR
